guide/src/module.md

```diff
@@ -105,3 +105,49 @@ submodules by using `from parent_module import child_module`. For more informati
 [#1517](https://github.com/PyO3/pyo3/issues/1517#issuecomment-808664021).
 
 It is not necessary to add `#[pymodule]` on nested modules, which is only required on the top-level module.
+
+## Declarative modules (experimental)
+
+An other syntax based on Rust inline modules is also available to declare modules.
+The `experimental-declarative-modules` feature must be enabled to use it.
+
```

Comment:
Should we briefly note what might be subject to change before we call this ready? I guess mostly it is the submodule automatically going to `sys.modules` and also the setting of `#[pyclass(module = ....)]`.

We could also add a link to the issue.

(And I guess we thought we might change the attribute names, though I'm feeling quite happy about those.)

Reply:
Done: 6b31c398939662e77c06dc17803919ba229c1b98

BTW, sorry for my English, as you can see it's not my native language